### PR TITLE
osmocli: dec parser

### PR DESCRIPTION
**Pull request by RusAkh**
- add parser
- add test

Closes: #4092

## What is the purpose of the change

Adds `Dec` parser to `osmocli`. 

## Brief Changelog

ditto 

## Testing and Verifying

Adds test

## Documentation and Release Note

N/A

### CODE REVIEW

File: osmoutils/osmocli/parsers.go [hunk 1]
```diff
@@ -312,3 +316,12 @@ func ParseSdkInt(arg string, fieldName string) (sdk.Int, error) {
 	}
 	return i, nil
 }
+
+func ParseDec(arg, fieldName string) (types.Dec, error) {
+	fmt.Println(arg)
```
Comment (p0mvn) [inline osmoutils/osmocli/parsers.go:321]: Just to make sure this is removed before merge but please keep if it helps you debug before the PR is ready
```suggestion
```

File: osmoutils/osmocli/parsers.go [hunk 2]
```diff
@@ -312,3 +316,12 @@ func ParseSdkInt(arg string, fieldName string) (sdk.Int, error) {
...
+	i, err := sdk.NewDecFromStr(arg)
+	if err != nil {
+		return types.Dec{}, fmt.Errorf("could not parse %s as sdk.Coins for field %s: %w", arg, fieldName, err)
```
Comment (p0mvn) [inline osmoutils/osmocli/parsers.go:324]:
```suggestion
		return types.Dec{}, fmt.Errorf("could not parse %s as sdk.Dec for field %s: %w", arg, fieldName, err)
```

**Review by p0mvn — APPROVED** [review]
This makes sense to me. Have you tested that this solution was indeed the fix needed for e2e?

File: go.mod [hunk 3]
```diff
@@ -50,11 +50,9 @@ require (
 	github.com/docker/distribution v2.8.1+incompatible // indirect
 	github.com/felixge/httpsnoop v1.0.2 // indirect
 	github.com/google/btree v1.1.2 // indirect
-	github.com/hashicorp/go-uuid v1.0.2 // indirect
-	github.com/jhump/protoreflect v1.13.1-0.20220928232736-101791cb1b4c // indirect
 	github.com/kkHAIKE/contextcheck v1.1.3 // indirect
 	github.com/maratori/testableexamples v1.0.0 // indirect
-	github.com/onsi/ginkgo v1.16.4 // indirect
+	github.com/nxadm/tail v1.4.8 // indirect
```
Comment (p0mvn) [inline go.mod:55]: Should we be updating `osmoutils`?

**Review by ValarDragon — APPROVED** [review]

File: go.mod (hunk 3 above)
Comment (p0mvn) [reply go.mod:55]: This cone be done in this PR in progress: https://github.com/osmosis-labs/osmosis/pull/4038

**Pull request merged**